src/toil/test/cwl/cwlTest.py

```diff
@@ -451,6 +454,56 @@ def test_glob_dir_bypass_file_store(self) -> None:
             except FileNotFoundError:
                 pass
 
+    @needs_slurm
+    def test_slurm_node_memory(self) -> None:
+        from toil.cwl import cwltoil
+
+        # Run the workflow. This will either finish quickly and tell us the
+        # memory we got, or take a long time because it requested a whole
+        # node's worth of memory and no nodes are free right now. We need to
+        # support both.
+
+        # And if we run out of time we need to stop the workflow gracefully and
+        # cancel the Slurm jobs.
+
+        main_args = [
+            f"--jobStore={self.jobStoreDir}",
+            # Avoid racing to toil kill before the jobstore is removed
+            "--clean=never",
+            "--batchSystem=slurm",
+            "--no-cwl-default-ram",
+            "--slurmDefaultAllMem=True",
+            "--outdir",
+            self.outDir,
+            os.path.join(self.rootDir, "src/toil/test/cwl/measure_default_memory.cwl"),
+            os.path.join(self.rootDir, "src/toil/test/cwl/empty.json"),
+        ]
+        try:
+            log.debug("Start test workflow")
+            child = subprocess.Popen(["toil-cwl-runner"] + main_args, stdout=subprocess.PIPE)
+            output, _ = child.communicate(timeout=60)
+        except subprocess.TimeoutExpired:
+            # The job didn't finish quickly; presumably waiting for a full node.
+            # Stop the workflow
+            log.debug("Workflow might be waiting for a full node. Stop it.")
+            subprocess.check_call(["toil", "kill", self.jobStoreDir])
+            # Wait another little bit for it to clean up, making sure to collect output in case it is blocked on writing
+            child.communicate(timeout=20)
+            # Kill it off in case it is still running
+            child.kill()
+            # Reap it
+            child.wait()
+            # The test passes
+            return
```

Comment:
I think that this `return` is a no-op?

Reply:
I think I put it there to make it clear that we intend to go there and end the test. But if you think it makes more sense without the `return` I'll drop it.